src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -168,21 +169,27 @@ def _make_compile_request(self, versioned_target, dependencies, external_libs_pr
 
     include_dirs = [self._include_dirs_for_target(dep_tgt) for dep_tgt in dependencies]
     include_dirs.extend(self._get_third_party_include_dirs(external_libs_product, dependencies))
-
     sources_and_headers = self.get_sources_headers_for_target(target)
+    compiler_option_sets = (self._compile_settings.native_build_step_settings
+                                .get_compiler_option_sets_for_target(target))
 
     return NativeCompileRequest(
       compiler=self._compiler,
       include_dirs=include_dirs,
       sources=sources_and_headers,
-      fatal_warnings=self._compile_settings.get_fatal_warnings_value_for_target(target),
+      fatal_warnings=(self._compile_settings
```

Comment:
The `fatal_warnings` should be removable here, I think?

Reply:
I left a comment on #6710 to do this if it's nontrivial to extract here.

Reply:
I added a deprecation version to the register method for this option, so we don't break anybody. However, this feature might be young enough to just rip it out?

Reply:
I was referring to removing the `fatal_warnings` field of `NativeCompileRequest` as that can be calculated from the `compiler_option_sets` and `NativeCompileRequest` isn't exposed externally, so we don't need to deprecate that.